apps/remix-ide/src/app/components/plugin-manager-component.js

```diff
@@ -102,6 +102,57 @@ class PluginManagerComponent extends ViewPlugin {
     return this.appManager.actives.includes(name)
   }
 
+  async registerFlattener (event) {
+    if (!this.appManager.isActive('solidity')) { await this.appManager.activatePlugin('solidity') }
+    const compiledSuccessfully = await this.call('solidity', 'compileFile', event)
+    const reporCompileIssue = async () => {
+      this.call('fileManager', 'open', event.path[0])
+      addToolTip('Cannot flatten the file. Please make sure it is compiling successfully.')
+      await this.call('menuicons', 'select', 'solidity')
+    }
+    setTimeout(async () => {
+      if (compiledSuccessfully) {
+        const res = await this.call('solidity', 'getCompilationResult')
+        if (!res) {
+          reporCompileIssue()
+        } else {
+          await this.call('menuicons', 'select', 'flattener')
+          const path = await this.call('flattener', 'flattenAndSave', res)
+          await this.call('fileManager', 'open', path)
+        }
+      } else {
+        reporCompileIssue()
+      }
+    }, 1000)
```

Comment:
why are we doing a timeout?

Reply:
because the immediate call was not working